### ISSUE TITLE
More detailed error messages

### ISSUE THREAD

**Issue by Overdrivr** [conan-io/conan#429]
During consumption of a module, if there is an error in the `conanfile.py`, the error message is not very detailed.

For instance, in the `copy` method, array of source directories are not supported so the following code will not be accepted:

```
self.copy("*.h", dst="include", src=["platform/MKL25Z4/include",
                                             "platform/"])
```
While consumming the package, the following error is obtained
```
ERROR: platform-kl25z: join() argument must be str or bytes, not 'list'
```
A detailed error message would be more desirable
```
ERR: Module XXX: parsing error conanfile.py : line XXX :  join() argument must be str or bytes, not 'list'
```

**Comment by memsharded** [comment]
The PR has been merged, so it is fixed now. Please try if possible when the next 0.12 release is done and close the issue if everything fine, or give feedback. Thanks!

**Comment by Overdrivr** [comment]
Awesome ! Will try it ASAP. Thanks

**Comment by memsharded** [comment]
Released in conan 0.11.1, please try and reopen if necessary
